### PR TITLE
Fix repair

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Repair scans through from slot 0 every iteration.

#### Summary of Changes
Scan starting from root instead, repair detached heads.

Fixes #

### CODE REVIEW

File: core/src/repair_service.rs [hunk 1]
```diff
@@ -62,50 +50,36 @@ impl RepairService {
...
+                for ((to, req), repair_request) in reqs {
                     if let Ok(local_addr) = repair_socket.local_addr() {
                         submit(
                             influxdb::Point::new("repair_service")
-                                .add_field("repair_slot", influxdb::Value::Integer(slot as i64))
-                                .to_owned()
                                 .add_field(
-                                    "repair_blob",
-                                    influxdb::Value::Integer(blob_index as i64),
+                                    "repair_request",
```
Comment (rob-solana) [inline core/src/repair_service.rs]: this is gonna muck with our dashboard ;)

File: core/src/blocktree.rs [hunk 2]
```diff
@@ -765,9 +793,9 @@ impl Blocktree {
     }
 
     fn is_detached_head(meta: &SlotMeta) -> bool {
```
Comment (rob-solana) [inline core/src/blocktree.rs]: "detached_heads" is possibly a bad naming choice, and here it looks to consume a pub fn that's the opposite sense

perhaps "orphans" would be better, i.e. slots without parents

File: core/src/blocktree.rs [hunk 3]
```diff
@@ -127,6 +131,7 @@ pub struct Blocktree {
     erasure_cf: ErasureCf,
     detached_heads_cf: DetachedHeadsCf,
     pub new_blobs_signals: Vec<SyncSender<bool>>,
+    pub root_slot: RwLock<u64>,
```
Comment (rob-solana) [inline core/src/blocktree.rs]: this bugger doesn't need synchronization with anything else in this struct?

File: core/src/repair_service.rs (hunk 1 above)
Comment (carllin) [reply core/src/repair_service.rs]: Also updated the dashboard

File: core/src/blocktree.rs (hunk 3 above)
Comment (carllin) [reply core/src/blocktree.rs]: it's accessed from both repair_service and replay_stage

Comment (rob-solana) [reply core/src/blocktree.rs]: only written by one thread, then?  and written after all inputs from which we derive root have been read, right?  just asking about internal consistency.

Comment (carllin) [reply core/src/blocktree.rs]: yeah only written by replay_stage after locktower has finalized a new root.

**Pull request merged**